### PR TITLE
Add helper text to Customization form

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

> Use the annotations `template.kubevirt.io/images` and `template.kubevirt.io/containerdisks` to provide helper text in the form storage section

## 🎥 Demo

Screenshot from 2022-04-12 17-10-29

### CODE REVIEW

File: src/views/catalog/customize/components/CustomizeSource/utils.ts [hunk 1]
```diff
@@ -52,3 +60,27 @@ export const getPVCSource = (
...
+  const dataVolumeTemplates = getTemplateVirtualMachineObject(template)?.spec?.dataVolumeTemplates;
+
+  return dataVolumeTemplates?.[0]?.spec?.storage?.resources?.requests?.storage;
+};
+
+export const getHTTPHelperText = (template: V1Template, t: TFunction) => {
+  const importUrls = getTemplateImportURLs(template);
+
+  if (importUrls)
+    return t('Enter URL to download (for example: {{importUrls}}', {
```
Comment (yaacov) [inline src/views/catalog/customize/components/CustomizeSource/utils.ts:74]: it's a list (same for registry), you should pick only one example out of the list ( first one? )

File: src/utils/resources/template/utils/selectors.ts [hunk 2]
```diff
@@ -52,6 +52,20 @@ export const getTemplateProviderName = (template: V1Template): string =>
 export const getTemplateSupportLevel = (template: V1Template): string =>
   getAnnotation(template, ANNOTATIONS.supportLevel);
 
+/**
+ * A selector that returns the containerDisks of a given template
+ * @param {V1Template} template - template
+ */
```
Comment (yaacov) [inline src/utils/resources/template/utils/selectors.ts:58]: missing return value and type

File: src/views/catalog/customize/components/CustomizeSource/utils.ts (hunk 1 above)
Comment (upalatucci) [reply src/views/catalog/customize/components/CustomizeSource/utils.ts:74]: Curious... The metadata attribute of template has the type `ObjectMetadata`.
```js
export declare type ObjectMetadata = {
    annotations?: {
        [key: string]: string;
    };
    ...
};
```
So every annotation has type `string` and not `string[]`. Do we have to change this type? Or maybe the annotation is a string formatted in a particular way?

File: src/utils/resources/template/utils/selectors.ts (hunk 2 above)
Comment (upalatucci) [reply src/utils/resources/template/utils/selectors.ts:58]: okay thanks

File: src/views/catalog/customize/components/CustomizeSource/utils.ts (hunk 1 above)
Comment (yaacov) [reply src/views/catalog/customize/components/CustomizeSource/utils.ts:74]: intersting, we will need to check that, did you find a template with this annotations ?

File: src/utils/resources/template/utils/selectors.ts [hunk 3]
```diff
@@ -52,6 +52,22 @@ export const getTemplateProviderName = (template: V1Template): string =>
...
+  getAnnotation(template, ANNOTATIONS.containerDisks);
+
+/**
+ * A selector that returns the import URLs of a given template
+ * @param {V1Template} template - template
+ * @returns array of import URLS
+ */
+export const getTemplateImportURLs = (template: V1Template): string | undefined =>
+  getAnnotation(template, ANNOTATIONS.importURLs);
+
```
Comment (yaacov) [inline src/utils/resources/template/utils/selectors.ts:70]: yes, it's a list encoded as a string with EOL seperators

```suggestion
 * @returns {[]string} array of import URLS
 */
export const getTemplateImportURLs = (template: V1Template): []string =>
  template.metadata?.annotations['template.kubevirt.io/images']?.split('\n') || [];

```

Comment (upalatucci) [reply src/utils/resources/template/utils/selectors.ts:70]: Okay thanks :-)

**Pull request merged**